### PR TITLE
Refactor for better reorg detection

### PR DESCRIPTION
**Pull request by kudlaj**

### CODE REVIEW

File: internal/core/processor/universal/processor_test.go [hunk 1]
```diff
@@ -167,121 +167,133 @@ func TestGetLastBlock(t *testing.T) {
...
+		name                       string
+		startingBlock              uint64
+		previousBlockHeader        *types.Header
+		previousBlockData          model.Block
+		blockHeader                *types.Header
+		blockData                  model.Block
+		discoverReturn             bool
+		updateReturn               map[string][]model.ERC721Transfer
+		expectedError              error
+		expectedTxCommit           int
```
Comment (zorjak) [inline internal/core/processor/universal/processor_test.go]: Can `commit `happen if non `nil` is expected? Maybe I am wrong, but I think it is not possible.

File: internal/core/processor/universal/processor_test.go [hunk 2]
```diff
@@ -167,121 +167,133 @@ func TestGetLastBlock(t *testing.T) {
...
-		lastBlockDB            model.Block
-		lastBlockDBError       error
-		previousBlockData      *types.Header
-		previousBlockDataError error
-		expectedError          error
+		name                       string
+		startingBlock              uint64
+		previousBlockHeader        *types.Header
+		previousBlockData          model.Block
+		blockHeader                *types.Header
```
Comment (zorjak) [inline internal/core/processor/universal/processor_test.go]: I think BlockHeader and BlockData probably both contain the same information. Can you check if both of them are needed?

**Review by zorjak — APPROVED** [review]

File: internal/core/processor/universal/processor_test.go (hunk 2 above)
Comment (kudlaj) [reply internal/core/processor/universal/processor_test.go]: yes, they are needed but the name is confusing. I will change it

File: internal/core/processor/universal/processor_test.go (hunk 1 above)
Comment (kudlaj) [reply internal/core/processor/universal/processor_test.go]: you mean non nil error? Yes, but only if the error happens in the commit. But I prefer to have two different variables for clarity.

**Review by AlexRipoll — APPROVED** [review]

**Pull request merged**